daml-assistant/daml-helper/src/DamlHelper.hs

```diff
@@ -532,8 +534,14 @@ withNavigator (SandboxPort sandboxPort) navigatorPort config args a = do
 -- | Whether `daml start` should open a browser automatically.
 newtype OpenBrowser = OpenBrowser Bool
 
-runStart :: OpenBrowser -> IO ()
-runStart (OpenBrowser shouldOpenBrowser) = withProjectRoot Nothing (ProjectCheck "daml start" True) $ \_ _ -> do
+-- | Whether `daml start` should start the navigator automatically.
+newtype StartNavigator = StartNavigator Bool
+
+-- | Whether `daml start` should wait for Ctrl+C or interrupt after starting servers.
+newtype WaitForSignal = WaitForSignal Bool
```

Comment:
What is the usecase for this?

Reply:
For when you want to run a script using `--on-start` and then quit after the script finishes (i.e. for testing).

Reply:
Thanks, sounds like a reasonable usecase. An alternative worth considering might be to exit if the start script fails with a non-zero exit code but I don’t feel strongly about this so your choice.

Reply:
Seems reasonable to me to make `daml start` exit with the exit code of the `--on-start` command whenever it's non-zero.